cmake/bintools/gnu/target_strip.cmake

```diff
@@ -0,0 +1,64 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# Construct a commandline suitable for calling the toolchain binary tools
+#  version of strip.
+#
+#  Usage:
+#    bintools_strip(
+#      RESULT_CMD_LIST    <List of commands to be executed, usually after build>
+#      RESULT_BYPROD_LIST <List of command output byproducts>
+#
+#      STRIP_ALL          <When present, remove relocation and symbol info>
+#      STRIP_DEBUG        <When present, remove debugging symbols and sections>
+#      STRIP_DWO          <When present, remove DWARF .dwo sections>
+#
+#      FILE_INPUT         <The input file>
+#      FILE_OUTPUT        <The output file>
+#    )
+macro(bintools_strip)
```

Comment:
Could this be a function?

That would help with scoping.

Reply:
I did originally implement it as functions, but for some reason the list append did not work with functions, so I had to make it macros instead.

Reply:
Try using PARENT_SCOPE to modify the output variable.

macro's should only be used when necessary as they don't support function-local variables.

Reply:
PARENT_SCOPE does not work for list append, but since we are now storing the resulting command in a separate variable, there is no need for the list append. 
I have changed the macros to functions, and replaced the list appends with sets.